flink-connectors/flink-connector-pulsar/src/main/java/org/apache/flink/connector/pulsar/source/util/SerializableRange.java

```diff
@@ -0,0 +1,114 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.connector.pulsar.source.util;
+
+import org.apache.flink.shaded.guava18.com.google.common.base.MoreObjects;
+import org.apache.flink.shaded.guava18.com.google.common.collect.ComparisonChain;
+
+import org.apache.pulsar.client.api.Range;
+
+import javax.validation.constraints.NotNull;
+
+import java.io.Externalizable;
+import java.io.IOException;
+import java.io.ObjectInput;
+import java.io.ObjectOutput;
+import java.util.Objects;
+
+/** SerializableRange. */
+public class SerializableRange implements Externalizable, Comparable<SerializableRange> {
```

Comment:
I guess it would suffice to be `Serializable` and implement
```
   private void writeObject(java.io.ObjectOutputStream out)
       throws IOException
   private void readObject(java.io.ObjectInputStream in)
       throws IOException, ClassNotFoundException;
  ```